File: src/main/java/com/itechart/demojavakotlin/service/UserService.java
package com.itechart.demojavakotlin.service;

import com.itechart.demojavakotlin.entity.UserEntity;
import com.itechart.demojavakotlin.exceptions.NoMoneyNoHoneyException;
import com.itechart.demojavakotlin.model.UserRequest;

import java.math.BigDecimal;
import java.util.UUID;

public interface UserService {

    void payTicket(final UUID id, final BigDecimal bill) throws NoMoneyNoHoneyException;

    UserEntity createUser(final UserRequest userRequest);

    UserEntity getByName(final String name);
}

File: kotlinpath/demojavakotlin/src/main/kotlin/by/itechart/demojavakotlin/service/UserService.kt
package by.itechart.demojavakotlin.service

import by.itechart.demojavakotlin.entity.UserEntity
import by.itechart.demojavakotlin.model.UserRequest
import java.math.BigDecimal
import java.util.*

interface UserService {

    fun payTicket(id: UUID, bill: BigDecimal)

    fun createUser(userRequest: UserRequest): UserEntity

    fun getByName(name: String): UserEntity

}
